addon/sidebar.js

```diff
@@ -22,31 +11,33 @@ function displayPage(url, desktop) {
   element("#desktop").checked = !!desktop;
 }
 
-function displayHome() {
+async function displayHome() {
   element("#browser-container").style.display = "none";
   element("#onboarding").style.display = "";
-  browser.windows.getCurrent({populate: true}).then((windowInfo) => {
-    let tabList = element("#tabs");
-    tabList.innerHTML = "";
-    for (let tab of windowInfo.tabs) {
-      let li = document.createElement("li");
-      let image = document.createElement("img");
-      image.src = tab.favIconUrl;
-      let anchor = document.createElement("a");
-      anchor.href = tab.url;
-      anchor.textContent = tab.title;
-      anchor.addEventListener("click", (event) => {
-        displayPage(event.target.href);
-        event.preventDefault();
-        return false;
-      });
-      anchor.prepend(image);
-      li.appendChild(anchor);
-      tabList.appendChild(li);
-    }
-  }).catch((error) => {
+  let windowInfo;
+  try {
+    windowInfo = await browser.windows.getCurrent({populate: true});
+  } catch (error) {
     console.error("Error getting window info:", error);
```

Comment:
This translation isn't quite right – failing to get the windowInfo should be fatal (and probably will be a fatal TypeError later). We might not need any try/catch in this function at all.